src/vsr/replica.zig

```diff
@@ -3868,10 +3922,19 @@ pub fn ReplicaType(
             };
         }
 
-        /// Starting from the latest journal entry, backfill any missing or disconnected headers.
-        /// A header is disconnected if it breaks the chain with its newer neighbor to the right.
-        /// Since we work back from the latest entry, we should always be able to fix the chain.
-        /// Once headers are connected, backfill any dirty or faulty prepares.
+        /// Repair. Each step happens in sequence — step n+1 executes when step n is done.
+        ///
+        /// 1. Advance the head op to `op_repair_max = min(op_checkpoint_trigger, commit_max)`.
+        ///    To advance the head op we request+await a SV. Either:
+        ///    - the SV's "hook" headers include op_checkpoint_trigger (if we are ≤1 wrap behind), or
+        ///    - the SV is too far ahead, so we will fall back from WAL repair to state transfer.
+        /// 2. Acquire missing or disconnected headers in reverse chronological order, backwards from
+        ///    op_repair_max.
```

Comment:
Hm, we go backwards from `self.op`. It feels like `op` might actually be higher than `op_repair_max`?

Reply:
It isn't clear here, but in the next PR we will enforce:
`op` never exceeds `op_repair_max`.

That is, `op <= op_checkpoint_trigger` — even if we learn of a higher op, we can't increase our head op beyond the trigger.

Reply:
`commit_max` however can go as high as it likes